packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx

```diff
@@ -4,30 +4,88 @@ import styled from 'styled-components/native';
 import type { StyledBaseInputProps } from './StyledBaseInput.d';
 import { getBaseInputStyles } from './baseInputStyles';
 
-const StyledNativeBaseInput = styled.TextInput<StyledBaseInputProps & { isFocused: boolean }>(
-  (props) => ({
-    ...getBaseInputStyles({
-      theme: props.theme,
-      isFocused: props.isFocused,
-      isDisabled: !props.editable,
-      validationState: props.validationState,
-      leadingIcon: props.leadingIcon,
-      prefix: props.prefix,
-      interactionElement: props.interactionElement,
-      suffix: props.suffix,
-      trailingIcon: props.trailingIcon,
-    }),
+type StyledComponentAutoCompleteAndroid =
+  | 'off'
+  | 'name'
+  | 'cc-csc'
+  | 'cc-exp'
+  | 'cc-exp-month'
+  | 'cc-exp-year'
+  | 'cc-number'
+  | 'email'
+  | 'username'
+  | 'password'
+  | 'postal-code'
+  | 'tel'
+  | undefined;
+
+const autoCompleteSuggestionTypeAndroid = {
+  none: 'off',
+  name: 'name',
+  email: 'email',
+  username: 'username',
+  password: 'password',
+  newPassword: 'password-new',
+  oneTimeCode: 'sms-otp',
+  telephone: 'tel',
+  postalCode: 'postal-code',
+  countryName: 'postal-address-country',
+  creditCardNumber: 'cc-number',
+  creditCardCSC: 'cc-csc',
+  creditCardExpiry: 'cc-exp',
+  creditCardExpiryMonth: 'cc-exp-month',
+  creditCardExpiryYear: 'cc-exp-year',
+} as const;
+
+const autoCompleteSuggestionTypeiOS = {
+  none: 'none',
+  name: 'name',
+  email: 'emailAddress',
+  username: 'username',
+  password: 'password',
+  newPassword: 'newPassword',
+  oneTimeCode: 'oneTimeCode',
+  telephone: 'telephoneNumber',
+  postalCode: 'postalCode',
+  countryName: 'countryName',
+  creditCardNumber: 'creditCardNumber',
+  creditCardCSC: 'none',
+  creditCardExpiry: 'none',
+  creditCardExpiryMonth: 'none',
+  creditCardExpiryYear: 'none',
+} as const;
+
+const StyledNativeBaseInput = styled.TextInput<
+  StyledBaseInputProps & {
+    isFocused: boolean;
+    autoCompleteType?: typeof autoCompleteSuggestionTypeAndroid[keyof typeof autoCompleteSuggestionTypeAndroid];
+  }
+>((props) => ({
+  ...getBaseInputStyles({
+    theme: props.theme,
+    isFocused: props.isFocused,
+    isDisabled: !props.editable,
+    validationState: props.validationState,
+    leadingIcon: props.leadingIcon,
+    prefix: props.prefix,
+    interactionElement: props.interactionElement,
+    suffix: props.suffix,
+    trailingIcon: props.trailingIcon,
   }),
-);
+}));
 
 export const StyledBaseInput = ({
   name,
   isRequired,
   isDisabled,
   handleOnChange,
   handleOnBlur,
+  keyboardReturnKeyType,
+  autoCompleteSuggestionType,
   ...props
-}: StyledBaseInputProps): ReactElement => {
+}: StyledBaseInputProps & {
+  keyboardReturnKeyType?: Exclude<StyledBaseInputProps['keyboardReturnKeyType'], 'enter'>;
+}): ReactElement => {
   const [isFocused, setisFocused] = useState(false);
```

Comment:
nit:
```suggestion
  const [isFocused, setIsFocused] = useState(false);
```

Reply:
this is removed in PR5 so not touching it in this PR